90% port to 1.11.2

williewillus opened the pull request:
Why do ports happen? A very deep question indeed. Easy steps:

1. Want to play the mod really badly
2. Have regex
3. Have an evening to burn
4. ???
5. Profit

---

Anyways, this is a mostly complete, direct (no functional changes besides bugs) port to 1.11.2.

Things that work:
* It compiles and runs
* MOST of the itemstack nulls fixed

Things that don't work:
* ~~All of the assets are still broken, but I hear you have a script?~~
* I probably missed a few nulls
* Some of the reflection calls may be broken (due to mapping shifts), but mcpbot isn't on 1.11.2 yet so I can't check them
* And as always any new gameplay stuff might need gameplay changes to accomodate.

This supersedes #2100 completely.

Also the huge line changes are from renaming stuff, actual changes were not that big (broken up by commits below)

williewillus commented:
Edit: renamed all the assets to lowercase via the magic of shell scripts so there's that now

TheWhiteWolves commented:
before vaz inevitably comes in and says it, remember to rebase it down to 1 commit when your done :stuck_out_tongue:

Nincodedo commented:
@TheWhiteWolves I think github has an option to do that automatically now when Vaz goes to merge if its enabled on the repo. https://github.com/blog/2141-squash-your-commits

TheWhiteWolves commented:
@Nincodedo so it does, it is giving me the option for it to be either squashed and merged or rebased and merged.

Not merged anything in a while as I'm unable to check anything myself anymore now that my computer can't load MC without crashing.

Myknakryu commented:
Seems stable..... Pls fix that twisted helmets.
http://i.imgur.com/VCTw7Fq.png
And.... Crash
I've build mod about few minutes ago....
Minecraft version: 1.11.2
Forge version: 2206
AutoRegLib version: 1.1-4
Baubles version: 1.4.2
JEI version: 4.2.2.219

Logs:
Creative tab
http://pastebin.com/t5bdPzSK
JEI
http://pastebin.com/UDFzfygR

What happen:
I was searching through Botania creative tab then game crashed.... Also crashes in JEI.

Myknakryu commented:
Well Gaia guardian boss works as it should... but boss name in health bar have broken name. Also Key of the King's Law is somewhat broken. Textures and sprites most of the time disappear but item works normally.
http://imgur.com/a/qrJ9B

quat1024 commented:
>There is also no particles and animation on texture Terrestrial Agglomeration Plate while making Terrasteel.

This actually happens in 1.10 as well... there's probably a bug for this right

Myknakryu commented:
I have few problems with latests changes.
Well, inventory disappears. I assume that there was too many particles but i'm not sure. Once on chat while testing there was message saying that openGL has overflowed(sorry not in the video). 
https://youtu.be/47SjDKFO6xc

Myknakryu commented:
Spectrolus does not work.

williewillus commented:
@Myknakryu works for me

Myknakryu commented:
Well maybe a lag.... But asembly Halo does not work.

Myknakryu commented:
I've tested Terrasteel helmet with every ancient will... Guess what... Every ancient will crashes game.

Crash logs in all cases of applying wills are very similar.
http://pastebin.com/DMTaKAfK

williewillus commented:
all the nulls :P keep finding em for me pls

Myknakryu commented:
Does petal apothecary works?

williewillus commented:
it should now

Myknakryu commented:
Items made by crafty crate when block is bellow it are deleted.

williewillus commented:
1. need to investigate
2. intended, read the book
3. fixed
4. need to investigate
5. ?
6. it only works if GoG is enabled
7. need to investigate
8. fixed
9. fixed
10. can't reproduce
11. fixed
12. fixed
13. need to investigate

williewillus commented:
btw, only let me know of stuff that is different from 1.10. if it's a bug in 1.10 as well make a separate issue ticket for it. this PR isn't a "fix all the bugs" PR, it's a port.

Myknakryu commented:
So... Game crashes when player is placing Corporea Crystal Cube... And game is saved when Crystal cube is placed... Game crashes and after loading world i see Corporea crystal cube... There is something with placing this thing.

williewillus commented:
fixed

Myknakryu commented:
So here are two videos:
First one showing few bugs
https://youtu.be/1BzC52Tvbcw
Second one to ask if ender air should work this way
https://youtu.be/qmErOnlrTUk

williewillus commented:
If it happens in 1.10 I don't want to hear about it on this ticket, make a separate one. otherwise will take a look

Myknakryu commented:
Well i'm going to post few bugs for 1.10.2 version... Most of things are also broken. Gonna post video in few minutes...

williewillus commented:
yes, but do them on other tickets. this is just for issues introduced in 1.11. thanks.

Myknakryu commented:
Ok here it is for 1.10.2... and i'm going to write new issue for 1.10.
https://youtu.be/WyidFKhXefg

williewillus commented:
make separate reports for every issue, don't just lump it all into one

Myknakryu commented:
I don't think that's good idea i've counted 6 bugs which i'm not sure with two of those... And i have only one video.

williewillus commented:
No one will be able to read your tickets issues if they are not clear, **independent**, and readable

Tamaized commented:
Aye, you want all reported bugs to be separate so it's easy to know whats already fixed, what needs to be fixed on, and yea, what each bug is, if you mesh them all together in a single vid it's hard to tell what all the bugs are and frankly, a waste of time to go and find the specific one in a video.

williewillus commented:
moved to 1.11 branch on this repo
